### DISCUSSION TITLE
Error after Registration submit: Oops, an error occurred! Code: 202402211433065c83bc0e

### DISCUSSION THREAD

**Discussion by tototonon** [derhansen/sf_event_mgt#1197] [General]
After filling out the registration form for a Event and clicking submit an error appears in the frontend, Oops, an error occurred! Code: 202402211433065c83bc0e.

This happens when the confirm Registration Event is triggered like => ?tx_sfeventmgt_pieventregistration[action]=confirmRegistration&tx_sfeventmgt_pieventregistration[controller]=Event&tx_sfeventmgt_pieventregistration[hmac]=0078ded3f85855d118d177d7d4182e63b1dce933&tx_sfeventmgt_pieventregistration[reguid]=33&cHash=2913281f1f610bd4ce1094101cee409b

I configured all plugins like described in the Documentation.
I can't figure out what is going wrong and how to fix it

thank you

**Comment by derhansen** [comment]
Please provide stack trace from TYPO3 logfile, else no help is possible.

**Reply by tototonon** [reply]
Thank you for your quick response.
I have not found any specific error from the log , but the error only occurs, if i set an email address in the notification settings of the registration plugin.
If i have no email inserted, then it's fine.

**Reply by derhansen** [reply]
I'm sorry, but I still can not help without any logs or similar. The whole registration process including email sending is covered by acceptance tests, so I doubt your problem has something to do with the extension.

Please check your TYPO3 logs for the given error code and evaluate, what caused the problem.

**Reply by tototonon** [reply]
If i disable autoconfirmation for user registrations, this error appears.

(1/1) TypeError

DERHANSEN\SfEventMgt\Service\Notification\AttachmentService::getObjectAttachments(): Argument #1 ($propertyNames) must be of type array, string given, called in /homepages/15/d994272184/htdocs/******/******/typo3conf/ext/sf_event_mgt/Classes/Service/Notification/AttachmentService.php on line 82
in /homepages/15/d994272184/htdocs/*****/********/typo3conf/ext/sf_event_mgt/Classes/Service/Notification/AttachmentService.php line 183


    /**
     * Returns the attachments from an object of all configured properties
     */
    protected function getObjectAttachments(array $propertyNames, AbstractEntity $object): array
    {
        $attachments = [];
        if (count($propertyNames) > 0) {
            foreach ($propertyNames as $propertyName) {

**Reply by tototonon** [reply]
and 
(1/1) Symfony\Component\Mailer\Exception\TransportException

Connection to "process /usr/sbin/sendmail -bs" has been closed unexpectedly.
in /homepages/15/d994272184/htdocs/clickandbuilds/typo3_src-12.4.11/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php line 87

            if ($metas['timed_out']) {
                throw new TransportException(sprintf('Connection to "%s" timed out.', $this->getReadConnectionDescription()));
            }
            if ($metas['eof']) {
                throw new TransportException(sprintf('Connection to "%s" has been closed unexpectedly.', $this->getReadConnectionDescription()));
            }
        }

        $this->debug .= sprintf('< %s', $line);

**Reply by tototonon** [reply]
Maybe due to misconfiguration, or restrictions of the Webserver, when attempt sending Mails?

**Reply by derhansen** [reply]
In `AttachmentService::getObjectAttachments()` you handover a string instead of an array, that is why the error occurs. So there is a problem with your TypoScript setup. Please verify, that `settings.notification.*.attachments.*.*` are always an array and not a string. 

See https://docs.typo3.org/p/derhansen/sf_event_mgt/7.4/en-us/ForAdministrators/EmailAttachments/Index.html - all DataTypes must be array.

**Reply by derhansen** [reply]
The example TS in the docs seems wrong. Will fix that with https://github.com/derhansen/sf_event_mgt/issues/1200

**Reply by tototonon** [reply]
thank you

**Reply by tototonon** [reply]
Is there a specific MAIL configuration to make, for the registration confirmation mail? Because it says "Anmeldung erfolgreich

Ihre Anmeldung war erfolgreich", also in the Events Module, clicking the sent notification button, it says email sent successfully, but no email is outgoing/incoming?
And the integrated symfony mailer throws now errors.

**Reply by derhansen** [reply]
No special config needed. The extensions works with configured TYPO3 mails settings, so if TYPO3 can send mails through the install tool, also the extension can.

I however set this issue as resolved now, since this is not an issue with the extension, but general TYPO3 support.

**Reply by tototonon** [reply]
Okay. Thank you for your help. Maybe we'll hear in another context.
